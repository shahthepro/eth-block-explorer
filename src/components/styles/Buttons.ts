import styled from 'styled-components';

interface IButtonStyleProps {
  small?: boolean
}

const ButtonStyle = styled.button<IButtonStyleProps>`
  background-color: rgba(0, 122, 255, 1);
  color: #fff;
  border: 0;
  padding: ${props => props.small ? '0.6rem' : '1rem'};
  font-size: ${props => props.small ? '0.8rem' : '1rem'};
  cursor: pointer;

  &:hover {
    background-color: rgba(0, 122, 255, 0.8);
  }

  &:disabled {
    background-color: #ccc;
    color: #888;
  }
`;

const PrimaryButton = styled(ButtonStyle)`
  background-color: rgba(0, 122, 255, 1);

  &:hover {
    background-color: rgba(0, 122, 255, 0.8);
  }
`;

const SecondaryButton = styled(ButtonStyle)`
  background-color: rgba(255, 122, 0, 1);
  &:hover {
    background-color: rgba(255, 122, 0, 0.8);
  }
`;

export {
  PrimaryButton,
  SecondaryButton
}