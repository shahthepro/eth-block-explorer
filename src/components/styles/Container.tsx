import styled from 'styled-components';

const Container = styled.div`
  width: 980px;
  margin: 0 auto;
  padding: 4rem 0;
  box-sizing: border-box;
  height: 100%;
  max-height: 100%;
  overflow: auto;
`;

export default Container;