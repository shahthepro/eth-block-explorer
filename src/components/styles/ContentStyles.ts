import styled from 'styled-components';

const RowWrapper = styled.div`
  display: flex;
  flex-direction: row;
  box-sizing: border-box;
  overflow: auto;
`;

const ColumnWrapper = styled.div`
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  overflow: auto;
`;

const Wrapper = styled(ColumnWrapper)`
  border: 1px solid #f0f0f0;
  box-sizing: border-box;
  overflow: hidden;
  height: 100%;
`;

const Header = styled.div`
  font-size: 1.2rem;
  font-weight: 100;
  padding: 0.8rem 1rem;
  border-bottom: 1px solid #f0f0f0;
  text-transform: uppercase;
  flex: auto 0 0;
  display: flex;
`;

const HeaderContent = styled.div`
  flex: auto 1 1;
`;

const HeaderActions = styled.div`
  flex: auto 0 0;
`;

interface IContentProps {
  padded?: boolean,
  fixedSize?: boolean
  noScroll?: boolean
}

const Content = styled.div<IContentProps>`
  flex: ${props => props.fixedSize ? 'auto 0 0' : ' auto 1 1'};
  overflow: ${props => props.noScroll ? 'hidden' : 'auto'};
  padding: ${props => props.padded ? '1rem' : 0};
`;


const ColumnStyles = styled.div`
  padding: 0.8rem;
  display: inline-block;
  box-sizing: border-box;
`;

const ColumnOneFifth = styled(ColumnStyles)`
  width: 20%;
`;

const ColumnOneFourth = styled(ColumnStyles)`
  width: 25%;
`;

const ColumnFourFifth = styled(ColumnStyles)`
  width: 80%;
`;

const ColumnThreeFourth = styled(ColumnStyles)`
  width: 75%;
`;

const ColumnFullWidth = styled(ColumnStyles)`
  width: 100%;
`;

const BorderedWrapper = styled.div`
  width: 100%;
  border: 1px solid #f0f0f0;
`;

interface IListItemProps {
  nonClickable?: boolean
};

const ListItemStyle = styled.div<IListItemProps>`
  font-size: 1rem;
  padding: 0.8rem 1rem;
  border-bottom: 1px solid #f0f0f0;
  transition: background-color 0.5s ease;
  color: #444;
  box-sizing: border-box;
  background-color: #fff;

  &:last-child {
    border: 0;
  }

  &:hover {
    background-color: ${props => props.nonClickable ? '#fff' : '#f0f0f0'};
    color: ${props => props.nonClickable ? '#444' : '#000'};
    cursor: ${props => props.nonClickable ? 'auto' : 'pointer'};
  }
`;

export {
  Wrapper,
  Header,
  Content,
  HeaderContent,
  HeaderActions,
  RowWrapper,
  ColumnWrapper,

  BorderedWrapper,
  ColumnOneFifth,
  ColumnOneFourth,
  ColumnFourFifth,
  ColumnThreeFourth,
  ColumnFullWidth,

  ListItemStyle
}