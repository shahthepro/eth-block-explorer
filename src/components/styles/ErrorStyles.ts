import styled from 'styled-components';

const MessageStyle = styled.div`
  padding: 1rem;
  background: white;
  margin: 1rem 0;
  border: 1px solid rgba(0, 0, 0, 0.05);
  border-left: 5px solid red;
  p {
    margin: 0;
    font-weight: 100;
  }
  strong {
    margin-right: 1rem;
  }
`;

const InfoMessage = styled(MessageStyle)`
  border-left: 5px solid rgb(0, 122, 255);
`;

const ErrorMessage = styled(MessageStyle)`
  border-left: 5px solid red;
`;

export {
  InfoMessage,
  ErrorMessage
}