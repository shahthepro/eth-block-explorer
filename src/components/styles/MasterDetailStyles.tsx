import styled from "styled-components";

const MasterDetailWrapper = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  background-color: #fff;
`;

const MasterContainer = styled.div`
  flex: 300px 0 0;
`;
const DetailsContainer = styled.div`
  flex: auto;
`;

export {
  MasterContainer,
  DetailsContainer,
  MasterDetailWrapper,
}