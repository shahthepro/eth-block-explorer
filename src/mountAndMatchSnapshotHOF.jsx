export default function (it, ReactDOM, mount) {
  return function (elementName, element) {
    it(`renders <${elementName}> without crashing`, () => {
      const div = document.createElement('div');
      ReactDOM.render(element, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    
    it(`matches <${elementName}> snapshot`, () => {
      const mountedElement = mount(element);
      expect(mountedElement).toMatchSnapshot();
    });
  }
}