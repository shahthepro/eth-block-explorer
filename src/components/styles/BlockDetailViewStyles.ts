import styled from "styled-components";
import { BorderedWrapper, ListItemStyle } from './ContentStyles';

const BlockDetailsWrapper = styled(BorderedWrapper)`
  overflow-y: auto;
  max-height: 140px;
`;

const TransactionsListWrapper = styled(BorderedWrapper)`
  overflow-y: auto;
  height: 100%;
`;

const BlockDetailsRow = styled.div`
  width: 100%;
  border-bottom: 1px solid #f0f0f0;
  &:last-child {
    border: 0;
  }
`;

interface ITransactionsListItemProps {
  nonClickable?: boolean,
  expanded?: boolean
}

const TransactionsListItem = styled(ListItemStyle)<ITransactionsListItemProps>`
  padding: 0.8rem 1rem;
  color: ${props => props.expanded ? '#000' : '#444'};
  overflow: hidden;

  .dropdown-arrow {
    height: 16px;
    transition: transform 0.3s ease;
    transform: rotateX(${props => props.expanded ? '180' : '0'}deg);
  }
  .expanded-view {
    ${props => props.expanded ? '' : 'border: 0;'}
    transition: height 0.3s ease, margin 0.3s ease, opacity 0.3s ease;
    height: ${props => props.expanded ? 'auto' : '0px'};
    background-color: #fff;
    color: #000;
    margin: ${props => props.expanded ? '10px 0' : '0' };
    opacity: ${props => props.expanded ? '1' : '0' };
  }
  &:hover {
    cursor: pointer;
    background-color: #fff;
  }
  &:hover .expanded-view {
    cursor: auto;
    background-color: #fff;
    color: #000;
  }
`;

export {
  BlockDetailsWrapper,
  BlockDetailsRow,

  TransactionsListWrapper,
  TransactionsListItem
}