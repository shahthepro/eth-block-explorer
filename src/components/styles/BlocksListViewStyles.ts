import styled from "styled-components";
import { ListItemStyle } from './ContentStyles';

interface IBlocksListViewItemProps {
  selected?: boolean
}

const BlocksListViewItem = styled(ListItemStyle)<IBlocksListViewItemProps>`
  background-color: ${props => props.selected ? '#f0f0f0' : '#fff'};
  color: ${props => props.selected ? '#000' : '#444'};
  .block-number {
    color: #3498db;
  }

  .timestamp {
    color: #77788f;
    font-size: 0.7rem;
  }
`;

export {
  BlocksListViewItem,
}