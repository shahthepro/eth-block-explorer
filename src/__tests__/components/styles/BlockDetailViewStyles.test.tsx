import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { BlockDetailsWrapper, BlockDetailsRow, TransactionsListItem, TransactionsListWrapper } from './../../../components/styles/BlockDetailViewStyles';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import mountAndMatchSnapshotHOF from '../../../../mountAndMatchSnapshotHOF';

Enzyme.configure({ adapter: new Adapter() })
const { mount } = Enzyme;

const mountAndMatchSnapshot = mountAndMatchSnapshotHOF(it, ReactDOM, mount);

mountAndMatchSnapshot('BlockDetailsWrapper', <BlockDetailsWrapper />);
mountAndMatchSnapshot('BlockDetailsRow', <BlockDetailsRow />);
mountAndMatchSnapshot('TransactionsListItem', <TransactionsListItem />);
mountAndMatchSnapshot('TransactionsListWrapper', <TransactionsListWrapper />);
