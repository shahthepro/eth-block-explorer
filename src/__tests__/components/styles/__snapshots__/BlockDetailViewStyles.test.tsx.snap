// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches <BlockDetailsRow> snapshot 1`] = `
<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-cSHVUG",
          "isStatic": true,
          "lastClassName": "fMLWXv",
          "rules": Array [
            "
        width: 100%;
        border-bottom: 1px solid #f0f0f0;
        &:last-child {
          border: 0;
        }
      ",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-cSHVUG",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-cSHVUG fMLWXv"
    />
  </StyledComponent>
</styled.div>
`;

exports[`matches <BlockDetailsWrapper> snapshot 1`] = `
<Styled(styled.div)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-fjdhpX",
          "isStatic": true,
          "lastClassName": "eXyvVf",
          "rules": Array [
            "
        width: 100%;
        border: 1px solid #f0f0f0;
      ",
            "
        overflow-y: auto;
        max-height: 140px;
      ",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-VigVT",
        ],
        "render": [Function],
        "styledComponentId": "sc-fjdhpX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-VigVT sc-fjdhpX eXyvVf"
    />
  </StyledComponent>
</Styled(styled.div)>
`;

exports[`matches <TransactionsListItem> snapshot 1`] = `
<Styled(styled.div)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-kAzzGY",
          "isStatic": false,
          "lastClassName": "jTPYbN",
          "rules": Array [
            "
        font-size: 1rem;
        padding: 0.8rem 1rem;
        border-bottom: 1px solid #f0f0f0;
        transition: background-color 0.5s ease;
        color: #444;
        box-sizing: border-box;
        background-color: #fff;
      
        &:last-child {
          border: 0;
        }
      
        &:hover {
          background-color: ",
            [Function],
            ";
          color: ",
            [Function],
            ";
          cursor: ",
            [Function],
            ";
        }
      ",
            "
        padding: 0.8rem 1rem;
        color: ",
            [Function],
            ";
      
        .dropdown-arrow {
          height: 16px;
          transition: transform 0.3s ease;
          transform: rotateX(",
            [Function],
            "deg);
        }
        .expanded-view {
          ",
            [Function],
            "
          transition: height 0.3s ease, margin 0.3s ease;
          height: ",
            [Function],
            ";
          background-color: #fff;
          color: #000;
          margin: ",
            [Function],
            ";
        }
        &:hover {
          cursor: pointer;
          background-color: #fff;
        }
        &:hover .expanded-view {
          cursor: auto;
          background-color: #fff;
          color: #000;
        }
      ",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-jTzLTM",
        ],
        "render": [Function],
        "styledComponentId": "sc-kAzzGY",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-jTzLTM sc-kAzzGY jTPYbN"
    />
  </StyledComponent>
</Styled(styled.div)>
`;

exports[`matches <TransactionsListWrapper> snapshot 1`] = `
<Styled(styled.div)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-jzJRlG",
          "isStatic": true,
          "lastClassName": "cZhRJb",
          "rules": Array [
            "
        width: 100%;
        border: 1px solid #f0f0f0;
      ",
            "
        overflow-y: auto;
        height: 100%;
      ",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-VigVT",
        ],
        "render": [Function],
        "styledComponentId": "sc-jzJRlG",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-VigVT sc-jzJRlG cZhRJb"
    />
  </StyledComponent>
</Styled(styled.div)>
`;
